Library/DunningCampaign.cs

```diff
@@ -0,0 +1,189 @@
+
+    public DateTime? DeletedAt { get; private set; }
+
+    internal const string UrlPrefix = "/dunning_campaigns/";
+
+    public List<string> PlanCodes { get; private set;}
+
+    #region Constructors
+
+    public DunningCampaign()
+    {
+    }
+
+    internal DunningCampaign(XmlTextReader xmlReader)
+    {
+      ReadXml(xmlReader);
+    }
+
+    #endregion
+
+    /// <summary>
+    /// Update the Dunning Campaign for multiple Plans
+    /// </summary>
+    public void BulkUpdate(List<string> planCodes)
+    {
+        PlanCodes = new List<string>();
+        foreach (string planCode in planCodes)
+        {
+          PlanCodes.Add(planCode);
+        }
+        // PUT /dunning_campaigns/<id>/bulk_update
+        Client.Instance.PerformRequest(Client.HttpRequestMethod.Put,
+            UrlPrefix + Uri.EscapeDataString(Id.ToString()) + "/bulk_update",
+            WriteBulkUpdateXml);
+    }
+
+    internal override void ReadXml(XmlTextReader reader)
+    {
+      while (reader.Read())
+      {
+        if (reader.Name == "dunning_campaign" && reader.NodeType == XmlNodeType.EndElement)
+          break;
+
+        if (reader.NodeType != XmlNodeType.Element) continue;
+
+        DateTime dt;
+
+        switch (reader.Name)
+        {
+          case "id":
+            Id = reader.ReadElementContentAsString();
+            break;
+
+          case "name":
+            Name = reader.ReadElementContentAsString();
+            break;
+
+          case "code":
+            Code = reader.ReadElementContentAsString();
+            break;
+
+          case "description":
+            Description = reader.ReadElementContentAsString();
+            break;
+
+          case "default_campaign":
+            DefaultCampaign = reader.ReadElementContentAsBoolean();
+            break;
+
+          case "dunning_cycles":
+            ReadXMLDunningCycles(reader);
+            break;
+
+          case "created_at":
+            CreatedAt = reader.ReadElementContentAsDateTime();
+            break;
+
+          case "updated_at":
+            UpdatedAt = reader.ReadElementContentAsDateTime();
+            break;
+
+          case "deleted_at":
+            if (DateTime.TryParse(reader.ReadElementContentAsString(), out dt))
+                DeletedAt = dt;
+            break;
+        }
+
+      }
+    }
+
+    internal void ReadXMLDunningCycles(XmlTextReader reader) {
+      DunningCycles.Clear();
+
+      while (reader.Read())
+      {
+        if (reader.Name == "dunning_cycles" && reader.NodeType == XmlNodeType.EndElement)
+          break;
+
+        if (reader.NodeType == XmlNodeType.Element && reader.Name == "dunning_cycle")
+        {
+            DunningCycles.Add(new DunningCycle(reader));
+        }
+      }
+    }
+
+    internal void WriteBulkUpdateXml(XmlTextWriter xmlWriter)
+    {
+      xmlWriter.WriteStartElement("dunning_campaign");
+      xmlWriter.WriteStartElement("plan_codes");
+      foreach (string planCode in PlanCodes) {
```

Comment:
Nit: C# puts opening curly braces on a new line

```suggestion
      foreach (string planCode in PlanCodes)
      {
```

Reply:
fixed